modules/accounts.js

```diff
@@ -156,17 +158,40 @@ Accounts.prototype.setAccountAndGet = function(data, cb, tx) {
 		throw err;
 	}
 
-	library.logic.account.set(
-		address,
-		data,
-		err => {
-			if (err) {
-				return setImmediate(cb, err);
-			}
-			return library.logic.account.get({ address }, cb, tx);
-		},
-		tx
-	);
+	const task = t =>
+		new Promise((resolve, reject) => {
+			library.logic.account.set(
+				address,
+				data,
+				setAccountErr => {
+					if (setAccountErr) {
+						library.logger.error('Set account failed', setAccountErr);
```

Comment:
We should mention some detail of the account on which error occurred. That will be helpful for troubleshooting. 

> Set account failed for "<address>"